### DISCUSSION TITLE
How to animate dismiss layer

### DISCUSSION THREAD

**Discussion by tianjos** [unpoly/unpoly#585] [Q&A]
Hello!!!

TLDR; how to animate dismiss layers?

I'm trying to set an animation on dismissed layer passing like below:

```
@!button.link({
                        'up-target': '[layout-main]',
                        'up-layer': 'new',
                        'up-animation': 'move-from-bottom',
                        'up-on-dismissed': 'up.animate("element", "move-to-bottom")'
                    })
```

But doesn't work has expected, I missed something?

Edited:
It's work with open/close animation config

```
up.layer.config.modal.openAnimation = 'move-from-bottom'
up.layer.config.modal.closeAnimation = 'move-to-bottom'
```

**Comment by lcmen** [comment]
@triskweline is there a way to customize a close animation for a new layer (the same way it's possible to do it for opening)? E.g. I would like to use a different (than default) close animation for a specific modal consitently on all dismiss controls (dismiss button, esc, outside click). Is it possible to configure dismiss animation with `up-layer` attribute?

**Reply by triskweline** [reply]
There's currently no attribute for it, but you can set it in an event handler like this:

```js
up.on('up:link:follow', '#my-special-link', ({ renderOptions}) => {
  renderOptions.closeAnimation = 'move-to-bottom'
})
```

This would affect all links with the ID `my-special-link`, but you can use any other selector.

**Reply by lcmen** [reply]
Thank you @triskweline for the snippet. It works for the links and buttons rendered within the modal.

Now, how can I add / change the animation when the modal is dismissed via backdrop click or ESC key on the keyboard?

**Reply by lcmen** [reply]
@triskweline is there a way to add / customize animation for backdrop click or ESC key?

**Reply by triskweline** [reply]
I don't think you can configure multiple close animations for a given overlay.

Maybe you can work around like this:

```js
up.on('up:layer:dismiss', (event) => {
  if (event.value === ':key') {
    event.preventDefault()
    event.layer.dismiss('my-escape-dismiss-animation')
  }
})

up.on('up:layer:dismiss', (event) => {
  if (event.value === ':outside') {
    event.preventDefault()
    event.layer.dismiss('my-backdrop-dismiss-animation')
  }
})
```

**Reply by lcmen** [reply]
Awesome, thank you so much!
